android/capacitor/src/main/java/com/getcapacitor/Bridge.java

```diff
@@ -194,10 +211,70 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
         }
       }
     });
+  }
 
+  private void loadCustomWebView() {
+    webView.setWebChromeClient(new BridgeWebChromeClient(this));
 
-    // Get to work
-    webView.loadUrl(appUrl);
+    // Saving off old client so no functionality is lost if
+    // custom webview implements an existing client
+    final WebViewClient oldClient = webView.getWebViewClient();
+    webView.setWebViewClient(new WebViewClient() {
+
+      @Override
+      public void onPageStarted(WebView view, String url, Bitmap favicon) {
+        if (oldClient != null) {
+          oldClient.onPageStarted(view, url, favicon);
+        }
+      }
+
+      @Override
+      public void onPageFinished(WebView view, final String location) {
+        if (oldClient != null) {
+          oldClient.onPageFinished(view, location);
+        }
+
+        JSInjector injector = getJSInjector();
+        injectScriptFile(view, injector.getScriptString());
```

Comment:
Think we could make this a feature of the webview client instead of needing overrides?

Reply:
I'm not sure what you mean by "feature of the webview client" 🙃